Doc/glossary.rst

```diff
@@ -289,6 +297,21 @@ Glossary
       advanced mathematical feature.  If you're not aware of a need for them,
       it's almost certain you can safely ignore them.
 
+   concurrency
+      The ability of different parts of a program to be executed out-of-order
+      or in partial order without affecting the outcome.  This allows for
+      multiple tasks to make progress during overlapping time periods, though
+      not necessarily simultaneously.  In Python, concurrency can be achieved
```

Comment:
@colesbury @lysnikolaou Perhaps:

```suggestion
      Different parts of a program may be executed in the same time period without affecting the outcome.  This allows for
      multiple tasks to make progress during the same time period, though
      not necessarily simultaneously.  In Python, concurrency can be achieved
```

Reply:
As in "The property of different parts of a program being able to executed at the same time" etc.?